presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -143,6 +151,25 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
         return planNodeOrderings.min(possibleJoinNodes).getPlanNode();
     }
 
+    public static JoinNode confidenceBasedBroadcast(JoinNode joinNode, Context context)
+    {
+        if (!mustPartition(joinNode)) {
+            if (confidenceBasedBroadcastingEnabled(context.getSession())) {
```

Comment:
I personally feel that this confidenceBasedBroadcastingEnabled change should not belong to this function, but the caller of this function.